Mathlib/GroupTheory/GroupAction/Hom.lean

```diff
@@ -13,201 +17,419 @@ import Mathlib.Algebra.Module.Basic
-* `MulSemiringActionHomClass F M R S` states that `F` is a type of bundled `R → S` homs
-  preserving the ring structure and scalar multiplication by `M`
+* `MulActionHomClass F φ X Y` states that `F` is a type of bundled `X → Y` homs
+  which are `φ`-equivariant
+* `DistribMulActionHomClass F φ A B` states that `F` is a type of bundled `A → B` homs
+  preserving the additive monoid structure and `φ`-equivariant
+* `SMulSemiringHomClass F φ R S` states that `F` is a type of bundled `R → S` homs
+  preserving the ring structure and `φ`-equivariant
 
 ## Notations
 
-* `X →[M] Y` is `MulActionHom M X Y`.
-* `A →+[M] B` is `DistribMulActionHom M A B`.
-* `R →+*[M] S` is `MulSemiringActionHom M R S`.
+* `X →ₑ[φ] Y` is `MulActionHom φ X Y`.
+* `A →ₑ+[φ] B` is `DistribMulActionHom φ A B`.
+* `R →ₑ+*[φ] S` is `MulSemiringActionHom φ R S`.
 
+* When `M = N` and `φ = MonoidHom.id M`, we provide the notation `X →[M] Y`
 -/
 
-set_option autoImplicit true
-
 assert_not_exists Submonoid
 
-variable (M' : Type*)
-variable (X : Type*) [SMul M' X]
-variable (Y : Type*) [SMul M' Y]
-variable (Z : Type*) [SMul M' Z]
-variable (M : Type*) [Monoid M]
-variable (A : Type*) [AddMonoid A] [DistribMulAction M A]
-variable (A' : Type*) [AddGroup A'] [DistribMulAction M A']
-variable (B : Type*) [AddMonoid B] [DistribMulAction M B]
-variable (B' : Type*) [AddGroup B'] [DistribMulAction M B']
-variable (C : Type*) [AddMonoid C] [DistribMulAction M C]
-variable (R : Type*) [Semiring R] [MulSemiringAction M R]
-variable (R' : Type*) [Ring R'] [MulSemiringAction M R']
-variable (S : Type*) [Semiring S] [MulSemiringAction M S]
-variable (S' : Type*) [Ring S'] [MulSemiringAction M S']
-variable (T : Type*) [Semiring T] [MulSemiringAction M T]
+section MulActionHom
+
+variable {M' : Type*}
+variable {M : Type*} {N : Type*} {P : Type*}
+variable (φ : M → N) (ψ : N → P) (χ : M → P)
+variable (X : Type*) [SMul M X] [SMul M' X]
+variable (Y : Type*) [SMul N Y] [SMul M' Y]
+variable (Z : Type*) [SMul P Z]
 
 /-- Equivariant functions. -/
 -- Porting note: This linter does not exist yet
 -- @[nolint has_nonempty_instance]
 structure MulActionHom where
   /-- The underlying function. -/
   protected toFun : X → Y
-  /-- The proposition that the function preserves the action. -/
-  protected map_smul' : ∀ (m : M') (x : X), toFun (m • x) = m • toFun x
-#align mul_action_hom MulActionHom
+  /-- The proposition that the function commutes with the actions. -/
+  protected map_smul' : ∀ (m : M) (x : X), toFun (m • x) = (φ m) • toFun x
 
 /- Porting note: local notation given a name, conflict with Algebra.Hom.GroupAction
  see https://github.com/leanprover/lean4/issues/2000 -/
 @[inherit_doc]
-notation:25 (name := «MulActionHomLocal≺») X " →[" M:25 "] " Y:0 => MulActionHom M X Y
+notation:25 (name := «MulActionHomLocal≺») X " →ₑ[" φ:25 "] " Y:0 => MulActionHom φ X Y
 
-/-- `SMulHomClass F M X Y` states that `F` is a type of morphisms preserving
-scalar multiplication by `M`.
+@[inherit_doc]
```

Comment:
after the docstring above is updated, it would be good if this weren't just `inherit_doc` but instead a new docstring is provided indicating that it is a special case. (or doesn't make reference to `φ` at all.)

Reply:
I have given more explicit docstrings